src/app/features/reporting/sql-report/sql-report.service.ts

```diff
@@ -185,4 +331,29 @@ export class SqlReportService {
         moment(to).format("YYYY-MM-DD")
     );
   }
+
+  getCsv(sqlData: SqlReportRow[]): any {
+    let deepestLevel: number = sqlData.reduce(
+      (previousValue, currentValue) =>
+        Math.max(previousValue, currentValue.level),
+      0,
+    );
+
+    let csv = "";
+
+    csv += "Name" + ",".repeat(deepestLevel + 1) + "Value\n";
```

Comment:
any particular reason why we don't use PapaParse library for this?

Are we sure values are always escaped properly to not break CSV formating / delimiters?

Reply:
Was not that easy to handle the nesting properly with Papa. Key and Value are properly escaped.